spec/app/domain/authentication/authn_k8s/validate_pod_request_spec.rb

```diff
@@ -0,0 +1,290 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+
+RSpec.describe Authentication::AuthnK8s::ValidatePodRequest do
+  let(:error_template) { "An error occured" }
+
+  let(:k8s_resolver) { double("K8sResolver") }
+
+  let(:account) { "SomeAccount" }
+
+  let(:host_id) { "HostId" }
+  let(:host_role) { double("HostRole", :id => host_id) }
+
+  let(:k8s_authn_container_name) { 'kubernetes/authentication-container-name' }
+  let(:host_annotation_1) { double("Annot1", :values => { :name => "first" }) }
+  let(:host_annotation_2) { double("Annot2") }
+
+  let(:host_annotations) { [ host_annotation_1, host_annotation_2 ] }
+
+  let(:host) { double("Host", :role => host_role,
+                              :annotations => host_annotations) }
+
+  let(:k8s_host_object) { "K8sHostObject" }
+  let(:k8s_host_namespace) { "K8sHostNamespace" }
+  let(:k8s_host) { double("K8sHost", :account => account,
+                                     :conjur_host_id => host_id,
+                                     :namespace => k8s_host_namespace,
+                                     :object => k8s_host_object) }
+
+  let (:container_name) { "ContainerName" }
+
+  let(:bad_service_name) { "BadMockService" }
+  let(:good_service_name) { "MockService" }
+
+  let(:good_webservice) { double("GoodWebservice") }
+
+  let(:spiffe_name) { "SpiffeName" }
+  let(:spiffe_namespace) { "SpiffeNamespace" }
+  let(:spiffe_id) { double("SpiffeId", :name => spiffe_name,
+                                       :namespace => spiffe_namespace) }
+  let(:pod_request) { double("PodRequest", :k8s_host => k8s_host,
+                                           :spiffe_id => spiffe_id) }
+
+  let(:dependencies) { { resource_repo: double(),
+                         k8s_resolver: k8s_resolver,
+                         k8s_object_lookup: double() } }
+
+  before(:each) do
+    allow(Resource).to receive(:[])
+      .with(host_id)
+      .and_return(host)
+
+    allow(Resource).to receive(:[])
+      .with("#{account}:webservice:conjur/authn-k8s/#{good_service_name}")
+      .and_return(good_webservice)
+    allow(Resource).to receive(:[])
+      .with("#{account}:webservice:conjur/authn-k8s/#{bad_service_name}")
+      .and_return(nil)
+
+  end
+
+  context "invocation" do
+    subject(:validator) { Authentication::AuthnK8s::ValidatePodRequest
+      .new(dependencies: dependencies) }
+
+    it 'raises WebserviceNotFound error when webservice is missing' do
+      allow(pod_request).to receive(:service_id).and_return(bad_service_name)
+
+      expected_message = "Webservice '#{bad_service_name}' wasn't found"
+      expect { validator.(pod_request: pod_request) }.to raise_error(Authentication::AuthnK8s::WebserviceNotFound,
```

Comment:
Can you line-break at the `.to` to make this line shorter?

```suggestion
      expect { validator.(pod_request: pod_request) }
        .to raise_error(Authentication::AuthnK8s::WebserviceNotFound, expected_message)
```